h/static/scripts/reducers/annotations.js

```diff
@@ -272,6 +272,17 @@ function annotationExists(state, id) {
   });
 }
 
+function findIDsForTags(state, tags) {
```

Comment:
> This should probably live next to findByTag and findByID?

This function takes the global state object for consistency with the other selector functions (in this module and others) which are exported from the module (see the functions just above this one). `findByTag` and `findByID` are local helpers.

> More generally, I'm still concerned that we keep adding O(N^2) lookup functions... 

In this PR, I'm just moving this function here from `WidgetController`.

Until recently, there was no single unique key for annotations, but now we guarantee that every annotation has a unique local tag as soon as it is added to the store, so we could now replace the annotations array with a `tag => Annotation` map and then maintain a separate `ID => tag` map for example.